sklearn/metrics/_pairwise_distances_reduction/_middle_term_computer.pyx.tp

```diff
@@ -25,12 +25,12 @@ from ...utils._cython_blas cimport (
 )
 from ...utils._typedefs cimport float64_t, float32_t, int32_t, intp_t
 
-# TODO: change for `libcpp.algorithm.fill` once Cython 3 is used
-# Introduction in Cython:
-#
-# https://github.com/cython/cython/blob/05059e2a9b89bf6738a7750b905057e5b1e3fe2e/Cython/Includes/libcpp/algorithm.pxd#L50 #noqa
-cdef extern from "<algorithm>" namespace "std" nogil:
-    void fill[Iter, T](Iter first, Iter last, const T& value) except + #noqa
+#Updated for Cython 3
+if parse_version(Cython.__version__) < parse_version('3.0.0'):
+   cdef extern from "<algorithm>" namespace "std" nogil: 
+     void fill[Iter, T](Iter first, Iter last, const T& value) except + #noqa 
+else:
+     from libcpp.algorithm cimport fill
```

Comment:
We do not want to have a disjunction like that. I think we better wait for Cython 3 to be used by default before considering this PR. My bad.